src/Leaflet.Renderer.SVG.Tile.js

```diff
@@ -39,6 +39,16 @@ L.SVG.Tile = L.SVG.extend({
 		}
 	},
 
+	removeFrom: function (map) {
+		if (this.options.interactive) {
+			for (var i in this._layers) {
+				var layer = this._layers[i];
+				delete this._map._targets[L.stamp(layer._path)];
```

Comment:
Thanks for sharing your thoughts here, I think this sounds like a good plan. But when you say `FeatureLayer` shouldn't subclass `Layer` - it doesn't (https://github.com/Leaflet/Leaflet.VectorGrid/blob/master/src/Leaflet.VectorGrid.js#L187), but you mean the `PointLayer` etc.? While I agree that the abstraction isn't nice, how would you go about it? (Maybe open a new issue to discuss that.) Anyway, I'm positive about re-architecting the internals of VectorGrid, they're a bit ad hoc as it is now.

I will have to slap myself, since like you say `addTo` already touches `this._map._targets`, and apparently I wrote that code :)